core/src/test/java/org/kiwix/kiwixmobile/core/dao/NewBookmarksDaoTest.kt

```diff
@@ -0,0 +1,270 @@
+    val query: Query<BookmarkEntity> = mockk(relaxed = true)
+    every { queryBuilder.build() } returns query
+    val bookmarkEntities: List<BookmarkEntity> = listOf(bookmarkEntity())
+    every { query.find() } returns bookmarkEntities
+    val bookmarkItem = BookmarkItem(bookmarkEntity())
+    assertThat(newBookmarksDao.getBookmarks(false, zimFileReader))
+      .isEqualTo(listOf(bookmarkItem))
+  }
+
+  @Test
+  fun `getCurrentZimBookmarksUrl returns list of bookmarkUrl`() {
+    val zimFileReader: ZimFileReader = mockk()
+    val queryBuilder: QueryBuilder<BookmarkEntity> = mockk()
+    every { box.query() } returns queryBuilder
+    val zimId = ""
+    every { zimFileReader?.id } returns zimId
+    every { queryBuilder.equal(BookmarkEntity_.zimId, zimFileReader?.id) } returns queryBuilder
+    every { queryBuilder.or() } returns queryBuilder
+    val zimName = ""
+    every { zimFileReader?.name } returns zimName
+    every { queryBuilder.equal(BookmarkEntity_.zimName, zimFileReader?.name) } returns queryBuilder
+    every { queryBuilder.order(BookmarkEntity_.bookmarkTitle) } returns queryBuilder
+    val query: Query<BookmarkEntity> = mockk(relaxed = true)
+    every { queryBuilder.build() } returns query
+    val bookmarkUrl = "bookmarkUrl"
+    val propertyQuery: PropertyQuery = mockk()
+    every { query.property(BookmarkEntity_.bookmarkUrl) } returns propertyQuery
+    val bookmarkUrlArray = arrayOf<String>(bookmarkUrl)
+    every { propertyQuery.findStrings() } returns bookmarkUrlArray
+    assertThat(newBookmarksDao.getCurrentZimBookmarksUrl(zimFileReader))
+      .isEqualTo(listOf(bookmarkUrl))
+  }
+
+  @Test
+  fun `getCurrentZimBookmarksUrl returns list of distinct bookmarkUrl`() {
+    val zimFileReader: ZimFileReader = mockk()
+    val queryBuilder: QueryBuilder<BookmarkEntity> = mockk()
+    every { box.query() } returns queryBuilder
+    val zimId = ""
+    every { zimFileReader?.id } returns zimId
+    every { queryBuilder.equal(BookmarkEntity_.zimId, zimFileReader?.id) } returns queryBuilder
+    every { queryBuilder.or() } returns queryBuilder
+    val zimName = ""
+    every { zimFileReader?.name } returns zimName
+    every { queryBuilder.equal(BookmarkEntity_.zimName, zimFileReader?.name) } returns queryBuilder
+    every { queryBuilder.order(BookmarkEntity_.bookmarkTitle) } returns queryBuilder
+    val query: Query<BookmarkEntity> = mockk(relaxed = true)
+    every { queryBuilder.build() } returns query
+    val bookmarkUrl = "bookmarkUrl"
+    val propertyQuery: PropertyQuery = mockk()
+    every { query.property(BookmarkEntity_.bookmarkUrl) } returns propertyQuery
+    val bookmarkUrlArray = arrayOf<String>(bookmarkUrl, bookmarkUrl)
+    every { propertyQuery.findStrings() } returns bookmarkUrlArray
+    assertThat(newBookmarksDao.getCurrentZimBookmarksUrl(zimFileReader))
+      .isEqualTo(listOf(bookmarkUrl))
+  }
+
+  @Test
+  fun `getCurrentZimBookmarksUrl returns empty list with empty zimName and zimId`() {
```

Comment:
`searches for empty name & id with a null zimFileReader` may be more accurate here